docs/datatype/image/text2image.md

```diff
@@ -1,3 +1,479 @@
+Read more about `match` [here](https://docs.jina.ai/fundamentals/document/documentarray-api/#matching-documentarray-to-another).
+
+
+```python
+from typing import Dict, Optional
+
+from jina import DocumentArray, Executor, requests
+from jina.types.arrays.memmap import DocumentArrayMemmap
+
+class SimpleIndexer(Executor):
+    """
+    A simple indexer that stores all the Document data together,
+    in a DocumentArrayMemmap object
+    To be used as a unified indexer, combining both indexing and searching
+    """
+
+    def __init__(
+        self,
+        match_args: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        Initializer function for the simple indexer
+        :param match_args: the arguments to `DocumentArray`'s match function
+        """
+        super().__init__(**kwargs)
+
+        self._match_args = match_args or {}
+        self._storage = DocumentArrayMemmap(
+            self.workspace, key_length=kwargs.get('key_length', 64)
+        )
+
+    @requests(on='/index')
+    def index(
+        self,
+        docs: Optional['DocumentArray'] = None,
+        **kwargs,
+    ):
+        """All Documents to the DocumentArray
+        :param docs: the docs to add
+        """
+        if docs:
+            self._storage.extend(docs)
+
+    @requests(on='/search')
+    def search(
+        self,
+        docs: Optional['DocumentArray'] = None,
+        parameters: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """Perform a vector similarity search and retrieve the full Document match
+        :param docs: the Documents to search with
+        :param parameters: the runtime arguments to `DocumentArray`'s match
+        function. They overwrite the original match_args arguments.
+        """
+        if not docs:
+            return
+        match_args = deepcopy(self._match_args)
+        if parameters:
+            match_args.update(parameters)
+
+        match_args = SimpleIndexer._filter_parameters(docs, match_args)
+
+        docs.match(self._storage, **match_args)
+
+    @staticmethod
+    def _filter_parameters(docs, match_args):
+        # get only those arguments that exist in .match
+        args = set(inspect.getfullargspec(docs.match).args)
+        args.discard('self')
+        match_args = {k: v for k, v in match_args.items() if k in args}
+        return match_args
+```
+
+## Building Flows
+### Indexing
+Now, after creating Executors, it's time to use them in order to build an index Flow and index our data.
+
+#### Building the index Flow
+We create a Flow object and add Executors one after the other with the right parameters: 
+
+1. CLIPImageEncoder: We specify the device. 
+2. SimpleIndexer: We need to specify the workspace parameter.
+
+```python
+from jina import Flow
+flow_index = Flow().add(uses=CLIPImageEncoder, name="encoder", uses_with={"device":device}) \
+      .add(uses=SimpleIndexer, name="indexer", workspace='workspace')
+flow_index.plot()
+```
```

Comment:
Can we format this so it has linebreaks? Right now it scrolls off the side of the screen